x/accounts/depinject.go

```diff
@@ -50,32 +44,19 @@ type ModuleOutputs struct {
 	Module         appmodule.AppModule
 }
 
-var _ signing.SignModeHandler = directHandler{}
-
-type directHandler struct{}
-
-func (s directHandler) Mode() signingv1beta1.SignMode {
-	return signingv1beta1.SignMode_SIGN_MODE_DIRECT
-}
-
-func (s directHandler) GetSignBytes(_ context.Context, _ signing.SignerData, _ signing.TxData) ([]byte, error) {
-	panic("not implemented")
-}
-
 func ProvideModule(in ModuleInputs) ModuleOutputs {
-	handler := directHandler{}
-	account := baseaccount.NewAccount("base", signing.NewHandlerMap(handler), baseaccount.WithSecp256K1PubKey())
-	accountskeeper, err := NewKeeper(
-		in.Cdc, in.Environment, in.AddressCodec, in.Registry, account,
-		accountstd.AddAccount(lockup.CONTINUOUS_LOCKING_ACCOUNT, lockup.NewContinuousLockingAccount),
-		accountstd.AddAccount(lockup.PERIODIC_LOCKING_ACCOUNT, lockup.NewPeriodicLockingAccount),
-		accountstd.AddAccount(lockup.DELAYED_LOCKING_ACCOUNT, lockup.NewDelayedLockingAccount),
-		accountstd.AddAccount(lockup.PERMANENT_LOCKING_ACCOUNT, lockup.NewPermanentLockingAccount),
-		accountstd.AddAccount(multisig.MULTISIG_ACCOUNT, multisig.NewAccount),
+	accCreators := make([]accountstd.AccountCreatorFunc, len(in.Accounts))
+	for i, acc := range in.Accounts {
```

Comment:
shouldn't we keep the baseapp as default always present account, so that the users only needs to provide options if he want to?

I cannot think of a reason of not adding the base account

Reply:
well i am not sure... because it would create dependency between accounts and base :S

Reply:
Ah, too bad. Makes sense then